src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,187 @@
+
+- [Upgrade to v4](/en/guides/upgrade-to/v4/)
+- [Upgrade to v3](/en/guides/upgrade-to/v3/)
+- [Upgrade to v2](/en/guides/upgrade-to/v2/)
+- [Upgrade to v1](/en/guides/upgrade-to/v1/)
+
+### Older docs (unmaintained)
+
+Documentation for older versions of Astro is not maintained, but is available as a static snapshot. Use these versions of docs if you are unable to upgrade your project, but still wish to consult guides and reference:
+
+- [unmaintained v3.6.3 snapshot](https://docs-git-v3-docs-unmaintained-astrodotbuild.vercel.app/)
+- [unmaintained v2.10.15 snapshot](https://deploy-preview-4405--astro-docs-2.netlify.app/en/getting-started/)
+
+
+## Semantic versioning
+
+Astro attempts to adhere as much as possible to [semantic versioning](https://semver.org/), which is a set of rules developers use to determine how to assign a version number to a release. Semantic version follows a predictable pattern to inform users of the kind of changes they can expect from one version to the next.
+
+Semantic versioning enforces a pattern of `X.Y.Z` for software version numbers. These values represent **major (X)**, **minor (Y)**, and **patch (Z)** updates.
+
+### Patch changes
+
+Patch changes are the least disruptive changes. They do not change the way you use Astro, and no change to your own code is required when you update.
+
+When Astro issues a "patch" version, the last number increases. (e.g. `astro@4.3.14` -> `astro@4.3.15`)
+
+Patches may be released for reasons such as:
+
+- Internal changes that do not change Astro’s functionality:
+  - refactors
+  - performance improvements
+  - increase or change in test coverage
+  - aligning with stated documentation and expected behavior
+- Improvements to logging and error messages.
+- Re-releases after a failed release.
+
+Patch changes also include **most bug fixes**, even in cases where users were taking advantage of existing unintended or undesirable behavior.
+
+
+### Minor changes
+
+Minor releases primarily introduce new features and improvements that you may wish to try, but require no changes to your code. Some existing features may also be **deprecated** (marked for deletion in a future version while continuing to function) in a minor release, giving you the opportunity to prepare for their eventual removal.
+
+Minor releases include changes such as:
+
+- **Deprecations** of existing features/options.
+- Introduction of new functionalities.
+- Upgrade of minimum required Node.js version (minor or patch), e.g. from `v18.4.0` to `v18.7.0`.
+  :::note
+  We take these kind of changes very seriously, and approve them only if we think that a new version of Node.js can improve the security or developer experience of our users, and impact core functionalities.
+  :::
+- Introduction of new options in the integration hooks.
+- Introduction of new functionalities in `astro/app`, notably used for creating new adapters.
+
+A minor release may also include smaller, patch changes at the same time.
```

Comment:
```suggestion
Minor releases include changes such as:

- **Deprecations** of existing features/options with a warning that they will be removed in an upcoming major release.
- Introduction of new functionalities.
- Introduction of new options in the integration hooks.
- Introduction of new functionalities in `astro/app`, notably used for creating new adapters.

:::note
Minor releases are meant to include only non-breaking changes.  However, *in rare cases*, Astro will include an upgrade of the minimum required Node.js version in a minor release (e.g. from `v18.4.0` to `v18.7.0`).

Astro is committed to adhering to semantic versioning, and will include an upgrade to a new version of Node.js in a minor release after careful analysis of a variety of factors, including the number of Astro projects running on incompatible versions, the security risk to developers of supporting an older version of Node.js, and the impact on Astro's core functionalities.
:::

A minor release may also include smaller, patch changes at the same time.
```

So, here's a suggested edit to start from! I'm not sure whether this is still too handwavy, BUT:
- I moved this rare edge case to the end, since we don't really need to stick something that almost never happens in the middle of a list
- I put the whole thing in the note, since the list doesn't have to be exhaustive (include changes such as...) and this is *all* kind of an aside. This gives you more flexibility in how you discuss it.